search/api.py

```diff
@@ -73,6 +81,59 @@ def gen_course_id(course_id):
     return "co_{}".format(safe_id)
 
 
+def gen_bootcamp_id(bootcamp_id):
+    """
+    Generates the Elasticsearch document id for a bootcamp
+
+    Args:
+        bootcamp_id (str): The course_id of a Bootcamp object
+
+    Returns:
+        str: The Elasticsearch document id for this object
+    """
+    safe_id = urlsafe_b64encode(bootcamp_id.encode("utf-8")).decode("utf-8").rstrip("=")
+    return "b_{}".format(safe_id)
+
+
+def gen_program_id(program_obj):
+    """
+    Generates the Elasticsearch document id for a program
+
+    Args:
+        program_obj (Program): The Program object
+
+    Returns:
+        str: The Elasticsearch document id for this object
+    """
+    safe_id = (
+        urlsafe_b64encode(program_obj.title.encode("utf-8"))
+        .decode("utf-8")
+        .rstrip("=")
+        .replace(" ", "_")
+    )
+    return "prog_{}".format(safe_id)
+
+
+def gen_learning_path_id(learning_path_obj):
+    """
+    Generates the Elasticsearch document id for a learning_path
+
+    Args:
+        learning_path_obj (LearningPath): The LearningPath object
+
+    Returns:
+        str: The Elasticsearch document id for this object
+    """
+    learning_path_id = str(learning_path_obj.author.username + learning_path_obj.title)
+    safe_id = (
+        urlsafe_b64encode(learning_path_id.encode("utf-8"))
+        .decode("utf-8")
+        .rstrip("=")
+        .replace(" ", "_")
+    )
+    return "lp_{}".format(safe_id)
```

Comment:
What should be the id for a Learning Path? Should we enforce uniqueness of titles for user-created Learning Paths? Should we allow same titles of LearningPaths across users and create a LearningPath id using a user's username and a title?

Reply:
As discussed in-person, we'll be using the autogenerated object ID. No uniqueness for titles.